metagraph/src/annotation/taxonomy/tax_classifier.cpp

```diff
@@ -0,0 +1,300 @@
+        auto it = accversion_to_taxid_map_.find(acc_version);
+        if (it == accversion_to_taxid_map_.end()) {
+            return false;
+        }
+        *taxid = it->second;
+        return true;
+    }
+
+    logger->error("Error: Could not get the taxid for label {}", label);
+    exit(1);
+}
+
+std::string TaxonomyBase::get_accession_version_from_label(const std::string &label) const {
+    if (label_type_ == TAXID) {
+        return utils::split_string(utils::split_string(label, "|")[2], " ")[0];
+    } else if (label_type_ == GEN_BANK) {
+        return utils::split_string(label, "|")[3];;
+    }
+
+    logger->error("Error: Could not get the accession version for label {}", label);
+    exit(1);
+}
+
+// TODO improve this by parsing the compressed ".gz" version (or use https://github.com/pmenzel/taxonomy-tools)
+void TaxonomyBase::read_accversion_to_taxid_map(const std::string &filepath,
+                                                const graph::AnnotatedDBG *anno_matrix = NULL) {
+    std::ifstream f(filepath);
+    if (!f.good()) {
+        logger->error("Error: Failed to open accession to taxid map table {}. \n"
+                      "In the cases when the taxid is not specified in the label string, "
+                      "the acc_version to taxid lookup table filepath must be given as a flag.", filepath);
+        exit(1);
+    }
+
+    std::string line;
+    getline(f, line);
+    if (!utils::starts_with(line, "accession\taccession.version\ttaxid\t")) {
+        logger->error("Error: The accession to taxid map table is not in the standard (*.accession2taxid) format {}",
+                      filepath);
+        exit(1);
+    }
+
+    tsl::hopscotch_set<std::string> input_accessions;
+    if (anno_matrix != NULL) {
+        for (const std::string &label : anno_matrix->get_annotation().get_all_labels()) {
+            input_accessions.insert(get_accession_version_from_label(label));
+        }
+    }
+
+    while (getline(f, line)) {
+        if (line == "") {
+            logger->error("Error: The accession to taxid map table contains empty lines. "
+                          "Please make sure that this file was not manually modified {}", filepath);
+            exit(1);
+        }
+        std::vector<std::string> parts = utils::split_string(line, "\t");
+        if (parts.size() <= 2) {
+            logger->error("Error: The accession to taxid map table contains incomplete lines. "
+                          "Please make sure that this file was not manually modified {}", filepath);
+            exit(1);
+        }
+        if (input_accessions.size() == 0 || input_accessions.count(parts[1])) {
+            accversion_to_taxid_map_[parts[1]] = std::stoul(parts[2]);
+        }
+    }
+}
+
+TaxonomyClsAnno::TaxonomyClsAnno(const graph::AnnotatedDBG &anno,
+                                 const double lca_coverage_rate,
+                                 const double kmers_discovery_rate,
```

Comment:
Don't pass simple types such as int/double/etc by const values. Usually, there is no difference and it doesn't help, so the more compact option is better.
```suggestion
                                 double lca_coverage_rate,
                                 double kmers_discovery_rate,
```